modules/mauth-test-utils/src/main/java/com/mdsol/mauth/test/utils/ProtocolTestSuiteHelper.java

```diff
@@ -0,0 +1,135 @@
+package com.mdsol.mauth.test.utils;
+
+import com.mdsol.mauth.test.utils.model.AuthenticationHeader;
+import com.mdsol.mauth.test.utils.model.SigningConfig;
+import com.mdsol.mauth.test.utils.model.UnsignedRequest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ProtocolTestSuiteHelper {
+
+  final static String TEST_SUITE_RELATIVE_PATH =
+      System.getenv("TEST_SUITE_RELATIVE_PATH") != null ? System.getenv("TEST_SUITE_RELATIVE_PATH")
+          : "../../../mauth-protocol-test-suite";
+  final static String MWSV2_TEST_CASE_PATH = TEST_SUITE_RELATIVE_PATH + "/protocols/MWSV2/";
+
+  public static SigningConfig loadSigningConfig() {
+    String configFile = TEST_SUITE_RELATIVE_PATH + "/signing-config.json";
+    ObjectMapper objectMapper = new ObjectMapper();
+    SigningConfig signingConfig = null;
+    try {
+      byte[] jsonData = Files.readAllBytes(normalizePath(configFile));
+      signingConfig = objectMapper.readValue(jsonData, SigningConfig.class);
+      return signingConfig;
+    } catch (IOException ex) {
+      System.out.println("Unable to load signing config file " + configFile);
+    }
+    finally {
+      return signingConfig;
+    }
+  }
+
+  public static String getPrivateKey(String keyFile) {
+    String filePath = TEST_SUITE_RELATIVE_PATH + keyFile;
+    try {
+      return new String(Files.readAllBytes(normalizePath(filePath)), Charset.defaultCharset());
+    } catch (IOException ex) {
+      throw new IllegalStateException("Unable to load the private key.", ex);
+    }
+  }
+
+  public static String getPublicKey() {
+    String filePath = TEST_SUITE_RELATIVE_PATH + "/signing-params/rsa-key-pub";
+    try {
+       return new String(Files.readAllBytes(normalizePath(filePath)), Charset.defaultCharset());
+    } catch (IOException ex) {
+      throw new IllegalStateException("Unable to load the public key." + filePath, ex);
+    }
+  }
+
+  public static byte[] loadTestData(String filePath) {
+    filePath = MWSV2_TEST_CASE_PATH + filePath;
+    byte[] bytes = new byte[0];
+    try {
+      bytes = Files.readAllBytes(normalizePath(filePath));
+    } catch (IOException ex) {
+      System.out.println("Unable to load: " + filePath);
+    }
```

Comment:
This needs to fail with an exception, not recover with an empty body......

Reply:
It is a little bit tricky here, this method is used to read file for all, currently some test cases may not include file for signer or authentication([see example](https://github.com/mdsol/mauth-protocol-test-suite/tree/develop/protocols/MWSV2/authentication-only-after-decoding)), so we just test Signer and/or Authentication if the related files are existing and can be loaded properly.

Reply:
Can we check that it exists before trying to load it, and not swallow the exception?

Or maybe we can have a clear separation of test type detection vs actual execution?
For example,

```
sealed trait TestCase

final case class TestWithStringBody(
  name: String,
  body: String,
  ...
) extends TestCase

final case class TestWithBinaryBody(
  name: String,
  body: Array[Byte]
  ...
)
```

That way we decouple the loading of the test cases from execution, and we're not swallowing exceptions which makes diagnosing test failures difficult.

I've also raised an issue in the test-suite repo about making test type detection easier
https://github.com/mdsol/mauth-protocol-test-suite/issues/4

Reply:
Would it be ok to merge as-is for now and have a follow-up story next sprint?

Reply:
@jatcwang thanks a lot, will do as suggested. I am going to implement "factory", so it will be reused easily for both core signing/authentication and httpclient integration

Reply:
@bgong-mdsol If you do it in Scala you can leverage `sealed trait` so you can get exhaustive matching in the tests

```
testSpecifications.foreach { testSpec =>
  testSpec match {
    case BinaryBodyTest(...) => 
    case AuthenticationOnlyTest(...) =>
    // if you have a new test case you'll get compile time warning that you're not handling it 
 }
}
```